library/objective-c/EnvoyEngine.h

```diff
@@ -7,6 +7,9 @@ NS_ASSUME_NONNULL_BEGIN
 /// A set of headers that may be passed to/from an Envoy stream.
 typedef NSDictionary<NSString *, NSArray<NSString *> *> EnvoyHeaders;
 
+/// A mutable set of headers that may be passed to/from an Envoy stream.
+typedef NSMutableDictionary<NSString *, NSArray<NSString *> *> EnvoyMutableHeaders;
```

Comment:
Shouldn't this be an `NSMutableArray` as the values?

Reply:
Sure, sounds good. I was trying to avoid unnecessary reallocations when converting from the immutable type, but honestly I wouldn't be surprised if Apple had a copy-on-write optimization under the hood anyways.